src/main/scala/cognite/spark/v1/FlexibleDataModelBaseRelation.scala

```diff
@@ -57,183 +58,162 @@ abstract class FlexibleDataModelBaseRelation(config: RelationConfig, sqlContext:
     })
   }
 
+  private def isReservedAttribute(instanceType: InstanceType, attribute: String) = {
+    val alwaysReservedAttributes: Set[String] = Set("space", "externalId", "_type")
+
+    // type is supported as an alias to _type for edges for legacy reasons.
+    val edgeReservedAttributes: Set[String] = Set("type", "startNode", "endNode")
+    alwaysReservedAttributes.contains(attribute) ||
+    (instanceType == InstanceType.Edge && edgeReservedAttributes.contains(attribute))
+  }
+
   private def extractInstancePropertyValue(key: String, value: InstancePropertyValue): Any =
     FlexibleDataModelRelationUtils.extractInstancePropertyValue(schema.apply(key).dataType, value)
 
-  protected def toInstanceFilter(
+  private def getVersionedViewExternalId(viewReference: ViewReference): String =
+    s"${viewReference.externalId}/${viewReference.version}"
+
+  private def getViewAttributeReference(
+      viewReference: ViewReference,
+      propertyName: String): Seq[String] =
+    Seq(viewReference.space, getVersionedViewExternalId(viewReference), propertyName)
+
+  protected def toFilter(
       instanceType: InstanceType,
       sparkFilter: Filter,
-      space: String,
-      versionedExternalId: String): Either[CdfSparkException, FilterDefinition] = {
+      // viewReference is required for non-reserved attribute filters resolution
+      viewReference: Option[ViewReference]
+  ): Either[CdfSparkException, FilterDefinition] =
+    sparkFilter match {
+      case EqualTo(attribute, _) if isReservedAttribute(instanceType, attribute) =>
```

Comment:
there's also `EqualNullSafe` and some others potentially applicable but let's stick to existing list of filters for now :+1: